src/libraries/System.Data.OleDb/src/OleDbComWrappers.cs

```diff
@@ -0,0 +1,107 @@
+
+        internal static OleDbComWrappers Instance { get; } = new OleDbComWrappers();
+
+        private OleDbComWrappers() { }
+
+        protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected override object CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+        {
+            Debug.Assert(flags == CreateObjectFlags.UniqueInstance);
+
+            Guid errorInfoIID = IID_IErrorInfo;
+            int hr = Marshal.QueryInterface(externalComObject, ref errorInfoIID, out IntPtr comObject);
+            if (hr == S_OK)
+            {
+                return new ErrorInfoWrapper(comObject);
+            }
+
+            throw new NotImplementedException();
+        }
+
+        protected override void ReleaseObjects(IEnumerable objects)
+        {
+            throw new NotImplementedException();
+        }
+
+        private class ErrorInfoWrapper : UnsafeNativeMethods.IErrorInfo, IDisposable
+        {
+            private readonly IntPtr _wrappedInstance;
+
+            public ErrorInfoWrapper(IntPtr wrappedInstance)
+            {
+                _wrappedInstance = wrappedInstance;
+            }
+
+            public void Dispose()
+            {
+                Marshal.Release(_wrappedInstance);
+            }
+
+            [Obsolete("not used", true)]
+            void UnsafeNativeMethods.IErrorInfo.GetGUID(/*deleted parameter signature*/)
+            {
+                throw new NotImplementedException();
+            }
+
+            public unsafe System.Data.OleDb.OleDbHResult GetSource(out string? source)
+            {
+                IntPtr pSource = IntPtr.Zero;
+                int errorCode = ((delegate* unmanaged<IntPtr, IntPtr*, int>)(*(*(void***)_wrappedInstance + 4 /* IErrorInfo.GetSource slot */)))
+                    (_wrappedInstance, &pSource);
+                if (pSource == IntPtr.Zero || errorCode < 0)
+                {
+                    source = null;
+                }
+                else
+                {
+                    source = Marshal.PtrToStringBSTR(pSource);
+                }
+
+                return (System.Data.OleDb.OleDbHResult)errorCode;
+            }
+
+            public unsafe System.Data.OleDb.OleDbHResult GetDescription(out string? description)
+            {
+                IntPtr pDescription = IntPtr.Zero;
+                int errorCode = ((delegate* unmanaged<IntPtr, IntPtr*, int>)(*(*(void***)_wrappedInstance + 5 /* IErrorInfo.GetDescription slot */)))
+                    (_wrappedInstance, &pDescription);
+                if (pDescription == IntPtr.Zero || errorCode < 0)
+                {
+                    description = null;
+                }
+                else
+                {
+                    description = Marshal.PtrToStringBSTR(pDescription);
```

Comment:
Similar to `GetSource()` above, this `BSTR` (i.e., `pDescription`) needs to be freed after conversion.